server/client_manager.py

```diff
@@ -288,7 +288,14 @@ def get_targets_by_ip(self, ip):
             if client.get_ip() == ip:
                 clients.append(client)
         return clients
-		
+
+    def get_targets_by_hdid(self, hdid):
+        clients = []
+        for client in self.clients:
+            if client.get_hdid() == hdid:
+                clients.append(client)
+        return clients
```

Comment:
Instead of declaring a list ahead of time, appending to it, and returning the list, you can just replace the line in the inner for loop that adds the client to the list to say `yield client`. Then you do not need a list or a return statement.